### ISSUE TITLE
Is it possible to make mpv transparent in macOS?

### ISSUE THREAD

**Issue by NightMachinery** [mpv-player/mpv#7147]
I have a usecase where I watch talks and exercise at the same time, and I like to put the exercise video on top of the talk, with some transparency applied. Unfortunately, system integrity protection and general Apple security practices mean that it is quite hard to get this behavior via a 3rd party window manager.
![image](https://user-images.githubusercontent.com/36224762/68673768-bdaa9700-0569-11ea-9eda-8e1575b9cd91.png)

**Comment by wm4** [comment]
Regarding alpha support in general, I don't know the OSX code; you can try the --alpha=yes option and see whether some random png appears transparent.

**Comment by Akemi** [comment]
i think there was some [code in the old cocoa backend](https://github.com/mpv-player/mpv/blob/master/video/out/opengl/context_cocoa.c#L139-L140) (the corresponding [commit](https://github.com/mpv-player/mpv/commit/d2baaaa7df87eba9566fb935eeab5bd0af80fe1b) from wm4) that made this possible? tbh i never tried it, also never had an example to test. i didn't implement this for the new cocoa-cb backend. though it could be done.

with a test file i could test it. also implementing this won't help with you use case since you would need a video with an alpha channel in the first place?

**Comment by orion1vi** [comment]
These properties should be of interest for achieving this:
https://developer.apple.com/documentation/appkit/nswindow/1419086-isopaque
https://developer.apple.com/documentation/appkit/nswindow/1419751-backgroundcolor
https://developer.apple.com/documentation/appkit/nsview/1483560-alphavalue

**Comment by Akemi** [comment]
that should be rather obvious from the PR i linked, besides the NSView one.

**Comment by NightMachinary** [comment]
True, it’s useful for videos with alpha channels only, while I need to
adjust the opacity of the whole mpv gui for any file.

**Comment by 2af** [comment]
Happy 2020, guys!
Any progress on this?)

**Comment by Akemi** [comment]
https://github.com/mpv-player/mpv/pull/17173 also added alpha support to `gpu-next` + `macvk`. just 'just' need to add an alpha channel to your video. something like this:

```
mpv --background=none --lavfi-complex="color=black@0.5:s=1920x1080,format=yuva420p,extractplanes=a [alpha];[vid1] format=yuv420p [yuv];[yuv][alpha] mergeplanes=0x00010210:yuva420p [vo]" VIDEO_FILE
```

could probably be done better with dynamic resolution for the alpha channel, since you need to match it to you video.

currently mpv/libplacebo uses only 2 alpha bits which is a bit too low for smooth alpha, though this will be addressed with https://code.videolan.org/videolan/libplacebo/-/merge_requests/767.

that's how it looked on my end:
<img width="1147" height="572" alt="Image" src="https://github.com/user-attachments/assets/ef1bc552-31c9-4219-81f7-6c4897bd790d" />
